### ISSUE TITLE
switch osx to cirun

### ISSUE THREAD

**Issue by h-vetinari** [conda-forge/pytorch-cpu-feedstock#429]
We've started timing out regularly on the osx builds; for the merge after #425, we needed 11(!) attempts for CI to pass (some agents on azure pipelines are faster than others, with no way to select between them).

There's some prior work on this in https://github.com/conda-forge/tensorflow-feedstock/pull/442, but we need to concurrently fix some stuff related to smithy and GHA labels for cirun: https://github.com/conda-forge/conda-smithy/issues/2324
